File: nodejs/examples/find-and-trigger-job.ts
/**
 * Example: Find a job by name (exact or partial) and trigger it.
 *
 * Prerequisites:
 *   - Set DATABRICKS_HOST and authentication env vars
 *   - Node.js >= 18
 */

import {
  DatabricksWorkspaceClient,
  AmbiguousJobError,
  ResourceNotFoundError,
} from "../src/index.ts";

const client = new DatabricksWorkspaceClient();
const job_name = 'jb_send_email_job_notification'
// Fire-and-forget by exact name
try {

  const result = await client.jobs.findAndTrigger({
    jobName: job_name,
    jobParameters: { env: "staging" },
// # or notebookParams : { env: "staging" }
  });
  console.log(`Triggered run ${result.runId} — state: ${result.lifecycleState}`);
} catch (err) {
  if (err instanceof ResourceNotFoundError) {
    console.log(`No job found matching: ${job_name}'`);
  } else if (err instanceof AmbiguousJobError) {
    console.log(`Multiple matches:\n${err.message}`);
  } else {
    throw err;
  }
}

// Wait for completion using a partial name
try {
  const result = await client.jobs.findAndTrigger({
    jobName: job_name, // partial match
    wait: true,
    timeoutMs: 30 * 60 * 1000,
    pollCallback: (s) =>
      console.log(`  [${s.lifecycleState}] ${s.stateMessage ?? ""}`),
  });
  console.log(`\nCompleted: ${result.resultState} (run ${result.runId})`);
} catch (err) {
  if (err instanceof AmbiguousJobError) {
    console.log(`Ambiguous match — narrow your search:\n${err.message}`);
  } else {
    throw err;
  }
}

// // Trigger by jobId (skips name resolution)
// const result = await client.jobs.findAndTrigger({ jobId: 123 });
// console.log(`Run ${result.runId}: ${result.lifecycleState}`);

File: python/examples/find_and_trigger_job.py
"""Example: Find a job by name (exact or partial) and trigger it.

Prerequisites:
    - Set DATABRICKS_HOST and authentication env vars
"""

from datetime import timedelta

from databricks_workspace_client import (
    AmbiguousJobError,
    DatabricksWorkspaceClient,
    ResourceNotFoundError,
)

client = DatabricksWorkspaceClient()

# Fire-and-forget by exact name
try:
    result = client.jobs.find_and_trigger(
        job_name="etl-daily",
        notebook_params={"env": "staging"},
    )
    print(f"Triggered run {result.run_id} — state: {result.life_cycle_state}")
except ResourceNotFoundError:
    print("No job found matching 'etl-daily'")
except AmbiguousJobError as e:
    print(f"Multiple matches:\n{e}")

# Wait for completion using a partial name
try:
    result = client.jobs.find_and_trigger(
        job_name="etl",  # partial match
        wait=True,
        timeout=timedelta(minutes=30),
        poll_callback=lambda s: print(f"  [{s.life_cycle_state}] {s.state_message or ''}"),
    )
    print(f"\nCompleted: {result.result_state} (run {result.run_id})")
except AmbiguousJobError as e:
    print(f"Ambiguous match — narrow your search:\n{e}")

# Trigger by job_id (skips name resolution)
result = client.jobs.find_and_trigger(job_id=123, wait=False)
print(f"Run {result.run_id}: {result.life_cycle_state}")
